### PR TITLE
common: use pthread_getname_np only if available

### PR DESCRIPTION
**Pull request by wjwithagen**
Signed-off-by: Willem Jan Withagen <wjw@digiware.nl>

### CODE REVIEW

File: src/common/assert.cc [hunk 1]
```diff
@@ -40,8 +40,10 @@ namespace ceph {
     g_assert_line = line;
     g_assert_func = func;
     g_assert_thread = (unsigned long long)pthread_self();
+#if defined(HAVE_PTHREAD_GETNAME_NP)
```
Comment (tchaikov) [inline src/common/assert.cc]: does this file `#include "include/compact.h"` indirectly? if yes, probably we can just call `ceph_pthread_getname()` here?

**Comment by tchaikov** [comment]
retest this please.

File: src/common/assert.cc [hunk 2]
```diff
@@ -12,6 +12,7 @@
  *
  */
 
+#include 'include/compat.h"
```
[the remark this answers is not in the document]
Comment (wjwithagen) [reply src/common/assert.cc]: @tchaikov 
Oeps, sorry about that.

[the remark this answers is not in the document]
Comment (wjwithagen) [reply src/common/assert.cc]: @tchaikov 
Ping??

**Pull request merged**